### PR TITLE
IRGen: Use new mangling for llvm type names.

### PR DESCRIPTION
**Pull request by eeckstein**
This should have no effect on the generated binary.

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci Please test and merge

**Comment by eeckstein** [comment]
@swift-ci Please test and merge

**Comment by eeckstein** [comment]
@swift-ci Please test and merge

**Comment by eeckstein** [comment]
@swift-ci Please smoke test and merge

**Comment by eeckstein** [comment]
@swift-ci Please smoke test and merge

**Comment by eeckstein** [comment]
@swift-ci Please smoke test linux

**Comment by eeckstein** [comment]
@swift-ci Please smoke test linux

**Comment by eeckstein** [comment]
@swift-ci Please smoke test linux

File: test/IRGen/struct_layout.sil [hunk 1]
```diff
@@ -3,21 +3,21 @@
...
 
 // 64: @_T04main14Rdar15410780_AVWV = hidden constant {{.*}} (i64 257
 // 64: @_T04main14Rdar15410780_BVWV = hidden constant {{.*}} (i64 258
 // 64: @_T04main14Rdar15410780_CVWV = hidden constant {{.*}} (i64 25
 
 
-// 32: %V4main14Rdar15410780_A = type <{ i2048, %Vs4Int8 }>
-// 32: %V4main14Rdar15410780_B = type <{ %GSqV4main14Rdar15410780_A_ }>
+// 32: %T4main14Rdar15410780_AV = type <{ i2048, %Ts4Int8V }>
+// 32: %T4main14Rdar15410780_BV = type <{ %GSqV4main14Rdar15410780_A_ }>
```
Comment (jrose-apple) [inline test/IRGen/struct_layout.sil]: You missed this one…

File: test/IRGen/struct_layout.sil [hunk 2]
```diff
@@ -3,21 +3,21 @@
...
 // 32: %GSqV4main14Rdar15410780_A_ = type <{ [257 x i8], [1 x i8] }>
-// 32: %V4main14Rdar15410780_C = type <{ %GSqSS_ }>
+// 32: %T4main14Rdar15410780_CV = type <{ %GSqSS_ }>
```
Comment (jrose-apple) [inline test/IRGen/struct_layout.sil]: …and this one.

**Pull request merged**